File: backend/app/schemas/email.py
from pydantic import BaseModel
from typing import Optional


class EmailResponse(BaseModel):
    id: str
    subject: str
    sender: str
    body: str
    summary: Optional[str] = None
    received_at: str
    is_read: bool = False
    priority: str = "medium"


class EmailSummaryResponse(BaseModel):
    email_id: str
    summary: str
    action_items: list = []

File: frontend/src/types/email.js
/**
 * Email object shape
 * @typedef {Object} Email
 * @property {string} id
 * @property {string} subject
 * @property {string} sender
 * @property {string} body
 * @property {string} summary
 * @property {string} receivedAt
 * @property {boolean} isRead
 * @property {string} priority - 'high' | 'medium' | 'low'
 */

export const EmailSample = {
  id: '',
  subject: '',
  sender: '',
  body: '',
  summary: '',
  receivedAt: '',
  isRead: false,
  priority: 'medium',
}
